source/FreeRTOS_IP_Utils.c

```diff
@@ -807,13 +809,17 @@ uint16_t usGenerateChecksum( uint16_t usSum,
 {
 /* MISRA/PC-lint doesn't like the use of unions. Here, they are a great
  * aid though to optimise the calculations. */
-    xUnion32 xSum2, xSum, xTerm;
+    xUnion32 xSum2;
+    xUnion32 xSum;
+    xUnion32 xTerm;
     xUnionPtr xSource;
     xUnionPtr xLastSource;
     uintptr_t uxAlignBits;
     uint32_t ulCarry = 0U;
     uint16_t usTemp;
     size_t uxDataLengthBytes = uxByteCount;
+    size_t ulSize;
+    size_t ulX;
```

Comment:
I think that you introduce `ulSize` and `ulX` for testing purposes? They do not really play a role here below.